dotnet/src/Connectors/Connectors.MistralAI/Client/MistralClient.cs

```diff
@@ -172,8 +174,9 @@ internal async Task<IReadOnlyList<ChatMessageContent>> GetChatMessageContentsAsy
 
                 // Now, invoke the function, and add the resulting tool call message to the chat options.
                 FunctionResult functionResult = new(function) { Culture = kernel.Culture };
-                AutoFunctionInvocationContext invocationContext = new(kernel, function, functionResult, chatHistory)
+                AutoFunctionInvocationContext invocationContext = new(kernel, function, functionResult, chatHistory, chatMessageContent)
```

Comment:
Why pass `chatMessageContent` separately?  Can't someone just read the last item from `chatHistory`?

Reply:
I think it was a request from SK users, but unfortunately, I don't have a use-case. @markwallace-microsoft do you have additional information about this?

Reply:
This was an ask from a customer. It's not obvious that the last message will be the correct one and we may not be able to guarantee this in the future so providing the associated chat message explicitly is a better contract.